### PR TITLE
Y axis label

### LINKED ISSUE #55: [Task] Add Y-Axis label vertically on side of graph
It should say "OP Allocation".
image

See this issue for more info on how this might be possible.
https://github.com/recharts/recharts/issues/184

### PR DESCRIPTION
**Pull request by swellander**
## Description
Adds `OP Allocation` vertical label to the Y-Axis

NOTE: based off of `vector-bar-graph` branch
Screenshot 2024-03-04 at 4 24 59 PM

## Related Issues
Closes #55

### CODE REVIEW

File: packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx [hunk 1]
```diff
@@ -121,7 +121,13 @@ export default function ImpactVectorGraph({ data }: { data: DataSet[] }) {
             tickMargin={10}
             scale={isLogarithmic ? logScale : "linear"}
             domain={["auto", "auto"]}
+            tickFormatter={abbreviateNumber}
```
Comment (swellander) [inline packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx:124]: Added this number abbreviation to reduce the overall width of the graph to make room for the new label. It's a little unconventional to do this for numbers above 1 million (`3000k` feels a bit strange), so I can try to find a different way to make space for the `OP Allocation` label if y'all want

File: packages/nextjs/utils/impactCalculator/common.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const abbreviateNumber = (num: number) => {
```
Comment (escottalexander) [inline packages/nextjs/utils/impactCalculator/common.ts:1]: Considering your comment about it seeming unconventional to have 3000k can we make this roll over to 1m when we get past 999k? Maybe use decimals for numbers between, like 1.25m, 1.6m etc. 

I like your approach with using abbreviations to make space for the label.

File: packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx [hunk 3]
```diff
@@ -121,7 +121,13 @@ export default function ImpactVectorGraph({ data }: { data: DataSet[] }) {
...
             allowDataOverflow
+            label={
+              <Text x={0} y={0} dx={15} dy={150} offset={0} angle={-90}>
```
Comment (escottalexander) [inline packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx:127]: Can we make the text a little bigger? 14px looks good to me.

**Review by escottalexander — CHANGES_REQUESTED** [review]
A couple small changes here. I like your idea about abbreviating the numbers. Saves space and looks better!

File: packages/nextjs/utils/impactCalculator/common.ts (hunk 2 above)
Comment (swellander) [reply packages/nextjs/utils/impactCalculator/common.ts:1]: Yeah great call

File: packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx (hunk 3 above)
Comment (swellander) [reply packages/nextjs/components/impact-vector/ImpactVectorGraph.tsx]: 👍 . I also moved it in a little bit closer to the axis. After abbreviating the higher numbers it felt like there was a bit too much space.
Screenshot 2024-03-06 at 6 58 00 AM

**Comment by swellander** [comment]
@escottalexander thanks for the reviews! I addressed your two comments

**Review by escottalexander — CHANGES_REQUESTED** [review]
Any idea why lower y-axis number aren't showing here? I don't see why it would be related but I feel like it must be. 
image

**Comment by swellander** [comment]
> Any idea why lower y-axis number aren't showing here? I don't see why it would be related but I feel like it must be.

Ok I think I've fixed it. It had to do with the `d3-scale` library that I had pulled in to fix the issue preventing us from viewing multiple vectors in log view. Not exactly sure why the linting check failed for the commit. Looks like the error was `Caused by: Error: connect ECONNREFUSED 127.0.0.1:8545`, but not sure why that would be happening after these changes 🤷. I might try force pushing these same changes to get CI to run again

**Comment by swellander** [comment]
> Not exactly sure why the linting check failed for the commit. Looks like the error was Caused by: Error: connect ECONNREFUSED 127.0.0.1:8545, but not sure why that would be happening after these changes 🤷.

Yeah so running CI on the same commit passed the second time. Strange

**Review by escottalexander — APPROVED** [review]
LGTM!

**Pull request merged**